src/Queues/MemoryQueue.php

```diff
@@ -63,21 +69,39 @@ public function post(Job &$job) {
      * Returns the number of job.
      * @return integer
      */
-    public function getQueueLength() {
+    public function getQueueLength()
+    {
         return count($this->_jobs);
     }
 
     /**
-     * 
+     * Empty the whole queue items.
+     * @return void
      */
-    public function emptyQueue() {
+    public function emptyQueue()
+    {
         $this->_jobs = [];
     }
     
     /**
+     * Returns the jobs posted to the queue.
      * @return Job[]
      */
-    public function getJobs() {
+    public function getJobs()
+    {
         return $this->_jobs;
     }
+
+    /**
+     * Release the job.
+     *
+     * @param Job $job The job to release.
+     * @return boolean whether the operation succeed.
+     */
+    protected function releaseJob(Job $job)
+    {
+        $this->_jobs[] = $job;
```

Comment:
since fetching the job from memory is basically deleting the job from the memory,, releasing it back does mean posting the job